### PR TITLE
doc: add CONTRIBUTING.md & update README.md

### PR DESCRIPTION
**Pull request by Hunam6**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Special thanks for your interest in contributing, you are more than welcome!
+
+## The translation from Go to V follow those steps:
```
Comment (JalonSolov) [inline CONTRIBUTING.md:3]:
```suggestion
## The translation from Go to V follows those steps:
```

Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
## The translation from Go to V follows those steps:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+1. Go2V CLI app
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
1. Start the process
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+   - `go2v.v`
+2. Get the Go code (path handling etc.)
```
Comment (JalonSolov) [inline CONTRIBUTING.md:7]:
```suggestion
2. The transpiler core starts by getting the Go code (path handling etc.)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	- `transpiler/go2v.v`
+3. Generate Go AST
+	- `transpiler/get_ast.go`
+	- `transpiler/go2v.v`
```
Comment (JalonSolov) [inline CONTRIBUTING.md:11]: Not necessary to show this again.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+4. Parse Go AST
+   - Tokenize Go AST
+      - `transpiler/tokenizer.v`
+   - Transform the list of tokens into a tree
+      - `transpiler/tree_constructor.v`
+5. Transform the parsed AST into a VAST struct
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
5. Transform the parsed Go AST into a V AST struct
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+   - `transpiler/v_style.v`
+7. Transform the V AST struct into V Code
+   - `transpiler/v_file_constructor.v`
+
+## When adding support for new cases you must add corresponding test(s).
+
+*In this section replace `go2v` with `v run .` or compile Go2V first.*
+
+Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file the supposed output.
+You can run the test suit using `go2v test` or specific one using `go2v test test_name`.
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
You can run the test suite using `go2v test` or specific one using `go2v test test_name`.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+- To create a test simply run the `go2v test -create test_name` command, it'll create a a dummy Go file at `tests/test_name/test_name.go`
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
- To create a test simply run `go2v test -create test_name` command, it'll create a dummy Go file at `tests/test_name/test_name.go`
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Hunam6) [reply CONTRIBUTING.md:11]: Okay maybe `transpiler/go2v.v` isn't useful here but IMHO `transpiler/get_ast.go` is useful

Comment (JalonSolov) [reply CONTRIBUTING.md:11]: I only meant the 2nd line.  :-)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+6. Adapt remaining Go-ish things to V-ish things
+   - `transpiler/v_style.v`
+7. Transform the V AST struct into V Code
+   - `transpiler/v_file_constructor.v`
+
+## When adding support for new cases you must add corresponding test(s).
+
+*In this section replace `go2v` with `v run .` or compile Go2V first.*
+
+Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file the supposed output.
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file containing the expected output.
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Hunam6) [reply CONTRIBUTING.md:11]: You want to remove "3. Generate Go AST"?

Comment (JalonSolov) [reply CONTRIBUTING.md:11]: I want to remove line 11:  "- `transpiler/go2v.v`" - that's the part that doesn't need to be repeated.  It's the second line in that section.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+   - `transpiler/v_style.v`
+7. Transform the V AST struct into V Code
+   - `transpiler/v_file_constructor.v`
+
+## When adding support for new cases you must add corresponding test(s).
+
+*In this section replace `go2v` with `v run .` or compile Go2V first.*
+
+Tests are located in `tests` subdirectories, each test is composed of a `.go` file, the input, and a `.vv` file containing the expected output.
+You can run the test suite using `go2v test` or specific one using `go2v test test_name`.
```
Comment (JalonSolov) [inline CONTRIBUTING.md]:
```suggestion
You can run the test suite using `go2v test` or a specific test using `go2v test test_name`.
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Hunam6) [reply CONTRIBUTING.md:11]: I don't really see why that's an issue, `transpiler/go2v.v` is effectively used in both steps

Comment (JalonSolov) [reply CONTRIBUTING.md:11]: Only to call the next routine.  It doesn't do the AST generation.  However, it doesn't really hurt anything to leave it, and if that's what you want to do, I have no objection.

**Pull request merged**